File: CrackingTheCodingInterview-Android/src/commonTest/kotlin/Chapter1/CheckPermutationSolverTests.kt
package Chapter1

import Chapter1.CheckPermutationSolver
import kotlin.test.Test
import kotlin.test.assertEquals

class CheckPermutationSolverTests {
    private val solver = CheckPermutationSolver()
    private val testCases: Array<Pair<Pair<String, String>, Boolean>> = arrayOf(
        Pair(Pair("AAC","CAA"), true),
        Pair(Pair("ABC","CAA"), false),
        Pair(Pair("AAC","AAC"), false),
    )

    @Test
    fun testUniqueWithMap() {
        for (testCase in testCases) {
            val result = solver.checkPermutationWithSort(testCase.first.first, testCase.first.second)
            assertEquals(result, testCase.second, "Hello")
        }
    }
}

File: CrackingTheCodingInterview-iOS/CrackingTheCodingInterviewTests/Chapter 1/CheckPermutationSolverTests.swift
//
//  CheckPermutationSolverTests.swift
//  CrackingTheCodingInterviewTests
//
//  Created by Kurt Jacobs on 2022/12/16.
//

import XCTest
@testable import CrackingTheCodingInterview

final class CheckPermutationSolverTests: XCTestCase {
    private let solver = CheckPermutationSolver()
    private let testCases: [(input: (str: String, str2: String), answer: Bool)] = [
        (("AAB","BAA"), true),
        (("AAC","AAC"), false),
        (("AABZ","BAAF"), false),
        (("AABA","ABAA"), true),
    ]

    func testPermutationSolver() throws {
        for testCase in testCases {
            let result = solver.checkPermutationWithSort(testCase.input.str, testCase.input.str2)
            XCTAssertEqual(result, testCase.answer)
        }
    }

    func testPermutationSolverPerformance() throws {
        self.measure {
            for testCase in testCases {
                solver.checkPermutationWithSort(testCase.input.str, testCase.input.str2)
            }
        }
    }

}
